app/(gcforms)/[locale]/(form administration)/form-builder/[id]/components/dialogs/RulesDialog/Rules.tsx

```diff
@@ -39,44 +38,53 @@ export const ModalFormRules = ({
     groupsEnabled: s.getGroupsEnabled(),
   }));
 
-  let sortedElements = sortByLayout({ layout: form.layout, elements: elements });
+  const initialChoiceRules = getElementsWithRuleForChoice({
+    formElements: elements,
+    itemId: item.id,
+  });
 
-  if (groupsEnabled) {
-    sortedElements = sortByGroups({ form: form, elements: elements });
+  if (mode === "add" && optionId) {
+    initialChoiceRules.push({ elementId: String(item.id), choiceId: optionId });
   }
 
   if (initialChoiceRules.length == 0) {
-    initialChoiceRules.push({ elementId: "", choiceId: `${item.id}.0` });
+    initialChoiceRules.push({ elementId: String(item.id), choiceId: `${item.id}.0` });
   }
 
-  const [choiceRules, setChoiceRules] = useState(initialChoiceRules);
+  const [choiceRules, setChoiceRules] = useState<ChoiceRule[]>(initialChoiceRules);
+
+  useEffect(() => {
+    choiceRulesRef.current = choiceRules;
+  }, [choiceRules, choiceRulesRef]);
+
+  let sortedElements = sortByLayout({ layout: form.layout, elements: elements });
+
+  if (groupsEnabled) {
```

Comment:
Thinking we don't need this check now thoughts ?

Reply:
nope d95ae58402224e375d78a2754d470cf4fe856d4e